### PR TITLE
Fetch fields from objects not the serialized hash, closes #67.

### LINKED ISSUE #67: Foreign keys for associations should not be required to include.
See: https://github.com/blakechambers/matterhorn/blob/master/lib/matterhorn/serialization/scoped/merge_inclusions.rb#L29

If FK is not in `items` which is JSON, then lookup can fail.

### PR DESCRIPTION
**Pull request by Nakort**

### CODE REVIEW

File: lib/matterhorn/links/relation/base.rb [hunk 1]
```diff
@@ -124,9 +124,11 @@ def find_with_ids(resource, ids)
           scope(resource).in(inverse_field_key => ids)
         end
 
-        def get_items_ids(items)
+        def get_items_ids(resource, items)
+          resource = [resource].flatten.inject({}){ |hsh, object| hsh[object[:_id]] = object; hsh}
```
Comment (blakechambers) [inline lib/matterhorn/links/relation/base.rb]: This needs to use the `ID_FIELD` constant.

Comment (blakechambers) [inline lib/matterhorn/links/relation/base.rb]: Merging this now would load the whole scope multiple times from the database (specifically one for each inclusion).  It should be memoized.

Can you add a spec asserting that we only load from db once?

Comment (Nakort) [inline lib/matterhorn/links/relation/base.rb]: I have moved the generation of the resource hash to be outside of the link generation so there is no need to memoize it now, how would you test it now that it's passed down?

**Comment by blakechambers** [comment]
The last thing I'm kinda against keeping, `find` currently passes serialized `items` into those methods, which isn't really required now that we are calling those methods on the resource itself.

So things like the entire `get_item_ids` method and how `find` is called from merge_inclusions should be removable.

**Comment by Nakort** [comment]
Agree, thanks for the catch, was caught up in keeping the old setup, much cleaner now.

**Pull request merged**